core/src/main/java/org/pentaho/di/core/database/AzureSqlDataBaseMeta.java

```diff
@@ -0,0 +1,113 @@
+/*! ******************************************************************************
+ *
+ * Pentaho Data Integration
+ *
+ * Copyright (C) 2002-2019 by Hitachi Vantara : http://www.pentaho.com
+ *
+ *******************************************************************************
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+package org.pentaho.di.core.database;
+
+import org.pentaho.di.core.exception.KettleDatabaseException;
+import org.pentaho.di.core.row.ValueMetaInterface;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+/**
+ * @author patitapaban19
+ *
+ */
+public class AzureSqlDataBaseMeta extends MSSQLServerDatabaseMeta {
+
+  public static final String JDBC_AUTH_METHOD = "jdbcAuthMethod";
+  public static final String IS_ALWAYS_ENCRYPTION_ENABLED = "azureAlwaysEncryptionEnabled";
+  public static final String CLIENT_ID = "azureClientSecretId";
+  public static final String CLIENT_SECRET_KEY = "azureClientSecretKey";
+
+  public static final String SQL_AUTHENTICATION = "SQL Server Authentication";
+  public static final String ACTIVE_DIRECTORY_PASSWORD = "Azure Active Directory - Password";
+  public static final String ACTIVE_DIRECTORY_MFA = "Azure Active Directory - Universal With MFA";
+
+
+  @Override
+  public String getDriverClass() {
+    if ( getAccessType() == DatabaseMeta.TYPE_ACCESS_ODBC ) {
+      return "sun.jdbc.odbc.JdbcOdbcDriver";
+    } else {
+      return "com.microsoft.sqlserver.jdbc.SQLServerDriver";
+    }
+  }
+
+  @Override
+  public String getURL( String hostname, String port, String databaseName ) {
+    if ( getAccessType() == DatabaseMeta.TYPE_ACCESS_ODBC ) {
+      return "jdbc:odbc:" + databaseName;
+    } else {
+      String url = "jdbc:sqlserver://" + hostname + ":1433;database=" + databaseName + ";encrypt=true;trustServerCertificate=false;hostNameInCertificate=*.database.windows.net;loginTimeout=30;";
```

Comment:
instead of the hard coded 1433 port, we would need to use the passed port right?